mindsdb/integrations/handlers/box_handler/box_handler.py

```diff
@@ -0,0 +1,258 @@
+                    ):
+                        has_content = True
+                        break
+                if has_content:
+                    print(df["path"])
+                    df["content"] = df["path"].apply(self._read_as_content)
+            else:
+                table = FileTable(self, table_name=table_name)
+                df = table.select(query)
+
+            return Response(RESPONSE_TYPE.TABLE, data_frame=df)
+        elif isinstance(query, Insert):
+            table_name = query.table.parts[-1]
+            table = FileTable(self, table_name=table_name)
+            table.insert(query)
+            return Response(RESPONSE_TYPE.OK)
+        else:
+            raise NotImplementedError(
+                "Only SELECT and INSERT operations are supported."
+            )
+
+    def get_tables(self) -> Response:
+        table_names = list(self._tables.keys())
+        df = pd.DataFrame(table_names, columns=["table_name"])
+        return Response(RESPONSE_TYPE.TABLE, data_frame=df)
+
+    def get_columns(self, table_name: str) -> Response:
+        table = self._get_table(Identifier(table_name))
+        columns = table.get_columns()
+        df = pd.DataFrame(columns, columns=["column_name"])
+        return Response(RESPONSE_TYPE.TABLE, data_frame=df)
+
+    def list_files_in_folder(self, folder_id, files=[]):
+        items = self.client.folders.get_folder_items(folder_id).entries
+        for item in items:
+            if item.type == 'folder':
+                self.list_files_in_folder(item.id)
+            elif item.type == 'file':
+                extension = item.name.split(".")[-1].lower()
+                if extension in self.supported_file_formats:
+                    file = self.client.files.get_file_by_id(item.id)
+                    full_path = [path.name for path in file.path_collection.entries][1:]
+                    full_path.append(item.name)
+                    files.append(
+                        {
+                            "path": "/".join(full_path),
+                            "name": item.name,
+                            "extension": extension,
+                        }
+                    )
+
+        return files
+
+    def _list_files(self, path=""):
+        return self.list_files_in_folder("0")
+
+    def _read_file(self, path) -> pd.DataFrame:
+        try:
+            buffer = self._read_as_content(path)
+            extension = path.split(".")[-1].lower()
+            if extension == "csv":
+                df = pd.read_csv(io.BytesIO(buffer))
+            elif extension == "tsv":
+                df = pd.read_csv(io.BytesIO(buffer), sep="\t")
+            elif extension == "json":
+                df = pd.read_json(io.BytesIO(buffer))
+            elif extension == "parquet":
+                df = pd.read_parquet(io.BytesIO(buffer))
+            else:
+                raise ValueError(f"Unsupported file format: {extension}")
```

Comment:
Maybe we include the info of the supported files too

Reply:
We are including the info of the supported files